protocols/smtp/src/main/java/org/apache/james/protocols/smtp/hook/HookResult.java

```diff
@@ -31,113 +35,81 @@
     private static final HookResult DENYSOFT = new HookResult(HookReturnCode.DENYSOFT);
     private static final HookResult DISCONNECT = new HookResult(HookReturnCode.DISCONNECT);
 
+    public static HookResult declined() {
+        return DECLINED;
+    }
+
+    public static HookResult ok() {
+        return OK;
+    }
+
+    public static HookResult deny() {
+        return DENY;
+    }
+
+    public static HookResult denysoft() {
+        return DENYSOFT;
+    }
+
+    public static HookResult disconnect() {
+        return DISCONNECT;
+    }
+
     private final int result;
     private final String smtpRetCode;
     private final String smtpDescription;
-    
-    /**
-     * Construct new HookResult
-     * 
-     * @param result 
-     * @param smtpRetCode 
-     * @param smtpDescription
-     */
+
     public HookResult(int result, String smtpRetCode, CharSequence smtpDescription) {
-        boolean match = false;
+        Preconditions.checkArgument(HookReturnCode.VALID_RETURN_CODE.contains(result),
```

Comment:
DISCONNECT was not treated the same as other code, isn't it ?

Reply:
Sorry, I fixed this in a later commit. Do you want me to correct that intermediary step?

Reply:
DONE